features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Operations/Operations.jsx

```diff
@@ -0,0 +1,444 @@
+        promisedResPolicies
+            .then((policies) => {
+                this.setState({ apiPolicies: policies.obj.list });
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
+                }
+                const { status } = error.status;
+                if (status === 404) {
+                    this.setState({ notFound: true });
+                } else if (status === 401) {
+                    doRedirectToLogin();
+                }
+            });
+    }
+
+    /**
+     *
+     * @param {*} event triggered for scope change
+     */
+    handleScopeChange(event) {
+        this.setState({ scopes: event.target.value });
+        this.handleScopeChangeInSwaggerRoot(event.target.value);
+    }
+
+    /**
+     *
+     * @param {*} scopes
+     */
+    handleScopeChangeInSwaggerRoot(scopes) {
+        const { swagger } = this.state.swagger;
+        if (swagger.security) {
+            swagger.security.map((object) => {
+                if (object.OAuth2Security) {
+                    object.OAuth2Security = scopes;
+                }
+                return object.OAuth2Security;
+            });
+        } else {
+            swagger.security = [{ OAuth2Security: scopes }];
+        }
+        this.setState({ swagger });
+    }
+
+    /**
+     *
+     * @param {*} newOperation
+     */
+    handleUpdateList(newOperation) {
+        const operationList = JSON.parse(JSON.stringify(this.state.operationList));
+        const updatedList = operationList.map(opertion => (opertion.target === newOperation.target
+            ? newOperation : opertion));
+        this.setState({ operationList: updatedList });
+    }
+
+    /**
+     *
+     */
+    updateOperations() {
+        const { api } = this.props;
+        const { operationList, intl } = this.state;
+        const apiObject = api._data;
+        apiObject.operations = operationList;
+        /* eslint no-underscore-dangle: ["error", { "allow": ["_data"] }] */
+        /* eslint no-param-reassign: ["error", { "props": false }] */
+        if (api._data) delete api._data;
+        if (api.client) delete api.client;
+        const promisedApi = this.newApi.update(api);
+        promisedApi
+            .then(() => {
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Details.Operations.Operations.api.updated.successfully',
+                    defaultMessage: 'API updated successfully!',
+                }));
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') { console.log(error); }
+                const { status } = error.status;
+                if (status === 404) {
+                    this.setState({ notFound: true });
+                }
+            });
+    }
+    toggleAssignScopes = () => {
+        this.setState({ showScopes: !this.state.showScopes });
```

Comment:
this can be done this way 

`
   this.setState(prevState => {
       return {showScopes: !prevState. showScopes}
    })
`